### DISCUSSION TITLE
Improve product combinations through price supplements

### DISCUSSION THREAD

**Discussion by alfonsolr09** [PrestaShop/PrestaShop#40172] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Let's imagine the following:

I sell a lock which can have the different combinations:

1. Color (gold or silver)
2. Size (30-30, 30-35, 35-35, 30-40, 40-40, 35-40, 40-40, etc...)
3. Cam type (short cam (L13) or long cam (L15))
4. Reinforced or unreinforced

Ok, we have a product with a total of 56 combinations (in this case there are few, it would be necessary to put more sizes since for the example I have put few)

Prestashop generates 56 combinations for us to price each of them individually

### Describe the solution you'd like

Don't you think it would be a better option to add the possibility of creating a price supplement?
Let me explain:
The default price for the product is in gold color, size 30-30, long cam and without reinforcement. 

Now I tell prestashop that:
- the combination with reinforcement has a surcharge of 15€
- the size 35-35 (for example) has 5€ surcharge
- short cam has 2€ 
- and silver color 10€.

Up to here I think it is understood

Then if the client selects a 30-30 lock in silver color, short cam and without reinforcement, automatically presatshop will calculate the price through the surcharges, that is, it will cost [default + silver color (10€) + short cam (2€) + without reinforcement (0€) ]

**In this way, you would only have to put the prices individually for the measurements, color, reinforcement, etc. and PrestaShop will do the complicated task.**

### Additional context

this example is with locks, but it is applicable to clothes, wheels, computer components, cars (with the extras for example), and millions of articles

In my case, I have come to collect more than 200 combinations between measurements, color, type of cam, reinforcement, knob, number of extra key copies, etc.. and management is impossible.

I attach a small video with an example of how it is now if you want to put several combinations

https://watch.screencastify.com/v/wraxZ7KFqH8RoER4xNIX

### Do you plan to work on this feature?

- [ ] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [ ] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by hibatallahAouadni** [comment]
Hello @alfonsolr09 

Thank you for your suggestion, I really like it :clap:  I'm thinking it will great if we add a new input **Impact on Price** in the attribute value form:

![image](https://user-images.githubusercontent.com/23308427/159270243-825b52d1-dd02-4911-9dd3-3e97190b7c49.png)

Could you please verify my mock-up and feedback.
The Product Team will take it into consideration for future developments. 
Please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you are willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help you by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know! 

Thank you

**Comment by alfonsolr09** [comment]
it's fine but instead of handling it from the attribute value form it would be better to handle it directly from the product combinations since the same attribute can have different prices depending on the product.

**Comment by hibatallahAouadni** [comment]
Hello @alfonsolr09 

Could you please provide a mockup for your proposal :pray: 

Thanks in advance!

**Comment by alfonsolr09** [comment]
@hibatallahAouadni 
![Imagen ](https://github.com/PrestaShop/PrestaShop/assets/62474824/8f793773-c00e-4d9a-93ee-e862b748551b)

It would be something like this seen from the frontpage.
An article with several combinations but instead of generating a lot of combinations you would only have to put them as a price supplement in the cost
